### PR TITLE
rework conflicting save data

### PR DESCRIPTION
**Pull request by z64a**
Allows loading vanilla saves and saves from other mods. These "invalid" saves display the mod name in red text instead of usual file information on the file select screen. Selecting one of these files shows a prompt with red text and "No" set as the default option. Valid saves behave normally.

### CODE REVIEW

File: src/battle/190B20.c [hunk 1]
```diff
@@ -643,20 +643,20 @@ void func_80263268(void) {
     }
 }
 
-void func_80263300(void) {
+void btl_init_menu_items(void) {
```
Comment (bates64) [inline src/battle/190B20.c:646]: Assuming all this stuff will get backported... can we make an issue or something to track these

File: src/dx/versioning.h [hunk 2]
```diff
@@ -26,14 +45,14 @@ typedef struct VanillaPlayerData {
     /* 0x008 */ s8 maxBP;
     /* 0x009 */ s8 level;
     /* 0x00A */ b8 hasActionCommands;
-    /* 0x00B */ char pad_00B;
+    /* 0x00B */ unsigned char pad_00B;
```
Comment (bates64) [inline src/dx/versioning.h:48]: Why?

**Review by bates64 — APPROVED** [review]

File: src/dx/versioning.h (hunk 2 above)
Comment (z64a) [reply src/dx/versioning.h:48]: Other unused/padding fields in structs are usually typed unsigned char, with signed char (arrays) only used for strings. This change is for consistency.

File: src/battle/190B20.c (hunk 1 above)
Comment (z64a) [reply src/battle/190B20.c:646]: Gonna diff enums, common_structs, and symbol_addrs.

**Pull request merged**